src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -409,7 +409,7 @@ protected boolean shouldExtractEntry( File targetDirectory, File targetFileName,
                 "" )
                 + suffix;
         boolean fileOnDiskIsNewerThanEntry = targetFileName.lastModified() >= entryDate.getTime();
-        boolean differentCasing = !entryName.equals( relativeCanonicalDestPath );
+        boolean differentCasing = !entryName.replace("/", File.separator).equals( relativeCanonicalDestPath );
```

Comment:
This change is wrong. Both strings would need to have "/" be replaced by File.separator if you want to look for different casing.